src/app/python/chat.py

```diff
@@ -0,0 +1,89 @@
+from threading import Lock
+
+
+class Chat(object):
+  """
+  Chat app module
+  """
+
+  def __init__(self, logger):
+    """
+    Chat initialiser
+
+    @type logger: logging.Logger
+
+    @param logger: The CIDE.py logger instance
+    """
+    self._logger = logger
+
+    self._usersLock = Lock()
+    self._users = set()
+
+    self._logger.debug("Chat module instance created")
+
+  def addUser(self, username):
+    """
+    Add a user to the chat
+
+    @param username: Username of the user to add
+
+    @return tuple (<<Data to send>>, <<Users to send to>>)
+    """
+    self._logger.info("Adding {0} to chat users".format(username))
+    data = {"author": "system", "message": username + " connected to the chat."}
+
+    with self._usersLock:
+      if username not in self._users:
+        self._users.add(username)
+        users = self._users.copy()
+
+      else:
+        users = []
+        self._logger.warning("{0} was already in chat users.".format(username))
+
+    return (data, users)
+
+  def removeUser(self, username):
+    """
+    Remove a user from the chat
+
+    @param username: Username of the user to remove
+
+    @return tuple (<<Data to send>>, <<Users to send to>>)
+    """
+    self._logger.info("Removing {0} from chat users".format(username))
+    data = {"author": "system", "message": username + " disconnected from the chat."}
+
+    with self._usersLock:
+      if username in self._users:
+        self._users.remove(username)
+        users = self._users.copy()
+
+      else:
+        users = []
+        self._logger.warning("{0} was not in chat users.".format(username))
+
+    return (data, users)
+
+  def handleMessage(self, author, message):
+    """
+    Handle the message. Return the data to send, and to who.
+
+    @param author: The author of the message
+    @param message: The message received
+
+    @return tuple (<<Data to send>>, <<Users to send to>>)
+    """
+    self._logger.debug("Handling message from {0}, message: {1}".format(author, message))
+    data = {"author": author, "message": message.strip()}
```

Comment:
@vphilippon : The following dictionnary appears many times:
 {"author": <author>, "message": <msg>} 

Could it be refactored into a method that return the dictionnary ?
Therefore, changing the structure params would be done at only one place.